ninja-core/src/main/java/ninja/ReverseRouter.java

```diff
@@ -0,0 +1,421 @@
+            }
+            return this;
+        }
+        
+        /**
+         * Add a list containing pairs with replacements for either path or query
+         * string parameters. This method will URL encode the values when building
+         * the final result.  When replacing path parameters it's recommended
+         * to use the <code>path</code> method since it will validate the parameter
+         * exists, while this method would fallback to adding the value as a query
+         * string parameter if the path parameter did not exist.
+         * @param parameterMap The list of pairs of name and values.
+         * @return A reference to this builder
+         * @see #path(java.lang.String, java.lang.Object)
+         * @see #query(java.lang.String, java.lang.Object) 
+         */
+        public Builder params(Object... parameterMap) {
+            return setParams(false, parameterMap);
+        }
+        
+        /**
+         * Identical to <code>params</code> except this method will NOT url
+         * encode the values when building the final result.  When replacing path
+         * parameters it's recommended to use the <code>path</code> method since
+         * it will validate the parameter exists, while this method would fallback
+         * to adding the value as a query string parameter if the path parameter
+         * did not exist.
+         * @param parameterMap The list of pairs of name and values.
+         * @return A reference to this builder
+         * @see #path(java.lang.String, java.lang.Object)
+         * @see #query(java.lang.String, java.lang.Object) 
+         */
+        public Builder rawParams(Object... parameterMap) {
+            return setParams(true, parameterMap);
+        }
+        
+        private Builder setParams(boolean raw, Object... parameterMap) {
+            if (parameterMap != null) {
+                if (parameterMap.length % 2 != 0) {
+                    throw new IllegalArgumentException("Always provide key (as String) value (as Object) pairs in parameterMap. That means providing e.g. 2, 4, 6... objects.");
+                }
+
+                for (int i = 0; i < parameterMap.length; i += 2) {
+                    setParam((String)parameterMap[i], parameterMap[i+1], raw);
+                }
+            }
+            return this;
+        }
+        
+        /**
+         * Add a parameter as a path replacement if it exists or will fallback
+         * to adding it a a query string value.  This method will URL encode the
+         * values when building the final result.
+         * @param name The parameter name
+         * @param value The parameter value
+         * @return A reference to this builder
+         * @see #path(java.lang.String, java.lang.Object)
+         * @see #query(java.lang.String, java.lang.Object) 
+         */
+        public Builder param(String name, Object value) {
```

Comment:
not sure - do we really need it? I am just thinking about how to save some lines of code...